src/uarch.rs

```diff
@@ -0,0 +1,380 @@
+use crate::uarch::Core::{Heterogeneous, Homogenous};
+use crate::Vendor;
+use crate::Vendor::{Intel, Amd};
+
+#[non_exhaustive]
+pub enum CoreArch {
```

Comment:
This seems to be sometimes using the "prototype" names for the platform and sometimes the "official" product names, is there some guidelines you followed for picking them?